### PR TITLE
Store batch xml payloads

### PR DESCRIPTION
**Pull request by kenfodder**
Doesn't interfere with the process just store the XML payloads and keep track of status and completion.

Todo:

- [x] Add extra metrics
- [x] Add tests for the XML response
- [x] Find a way to associate the individual updates the bulk update payload

### CODE REVIEW

File: db/migrate/20211104171214_create_cap_usage_update_payload.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateCapUsageUpdatePayload < ActiveRecord::Migration[6.1]
+  def change
+    create_table :cap_usage_update_payloads do |t|
+      t.string :payload_id
+      t.string :payload_xml
```
Comment (ltello) [inline db/migrate/20211104171214_create_cap_usage_update_payload.rb:5]: t.text?

**Comment by ltello** [comment]
Great!

File: db/migrate/20211104171214_create_cap_usage_update_payload.rb (hunk 1 above)
Comment (kenfodder) [reply db/migrate/20211104171214_create_cap_usage_update_payload.rb:5]: I checked that, it makes no difference...

https://www.postgresql.org/docs/9.1/datatype-character.html#:~:text=character%20without%20length%20specifier%20is,stores%20strings%20of%20any%20length.

**Review by markhallen — APPROVED** [review]
This looks really good. Thanks @kenfodder

File: spec/controllers/computacenter/api/cap_usage_controller_spec.rb [hunk 2]
```diff
@@ -139,6 +169,21 @@
...
+        payload = Computacenter::API::CapUsageUpdatePayload.order(id: :desc).limit(1).first
+
+        expect(payload.payload_id).to eq(payload_id)
+        expect(payload.payload_xml).to eq(cap_usage_update_packet)
+        expect(payload.payload_timestamp).to eq(payload_timestamp)
+        expect(payload.records_count).to eq(4)
+        expect(payload.succeeded_count).to eq(4)
+        expect(payload.failed_count).to eq(0)
+        expect(payload.status).to eq('succeeded')
+        expect(payload.completed_at).not_to be_nil
```
Comment (nigel-lowry) [inline spec/controllers/computacenter/api/cap_usage_controller_spec.rb:185]: could have `.to be_present`

File: spec/controllers/computacenter/api/cap_usage_controller_spec.rb [hunk 3]
```diff
@@ -187,6 +232,21 @@
...
+        payload = Computacenter::API::CapUsageUpdatePayload.order(id: :desc).limit(1).first
+
+        expect(payload.payload_id).to eq(payload_id)
+        expect(payload.payload_xml).to eq(cap_usage_update_packet)
+        expect(payload.payload_timestamp).to eq(payload_timestamp)
+        expect(payload.records_count).to eq(4)
+        expect(payload.succeeded_count).to eq(0)
+        expect(payload.failed_count).to eq(4)
+        expect(payload.status).to eq('failed')
+        expect(payload.completed_at).not_to be_nil
```
Comment (nigel-lowry) [inline spec/controllers/computacenter/api/cap_usage_controller_spec.rb:248]: ditto

File: spec/controllers/computacenter/api/cap_usage_controller_spec.rb [hunk 4]
```diff
@@ -199,6 +259,21 @@
...
+        payload = Computacenter::API::CapUsageUpdatePayload.order(id: :desc).limit(1).first
+
+        expect(payload.payload_id).to eq(payload_id)
+        expect(payload.payload_xml).to eq(cap_usage_update_packet)
+        expect(payload.payload_timestamp).to eq(payload_timestamp)
+        expect(payload.records_count).to eq(4)
+        expect(payload.succeeded_count).to eq(2)
+        expect(payload.failed_count).to eq(2)
+        expect(payload.status).to eq('partially_failed')
+        expect(payload.completed_at).not_to be_nil
```
Comment (nigel-lowry) [inline spec/controllers/computacenter/api/cap_usage_controller_spec.rb:275]: ditto

**Review by nigel-lowry — APPROVED** [review]

**Review by nigel-lowry — APPROVED** [review]

**Pull request merged**